fearless/Common/Storage/EntityToModel/AssetModelMapper.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+import RobinHood
+import CoreData
+
+enum AssetModelMapperError: Error {
+    case requiredFieldsMissing
+}
+
+class AssetModelMapper: CoreDataMapperProtocol {
+    var entityIdentifierFieldName: String { "id" }
+
+    func transform(entity: CDAsset) throws -> AssetModel {
+        guard let id = entity.id, let chainId = entity.chainId else {
+            throw AssetModelMapperError.requiredFieldsMissing
+        }
+
+        return AssetModel(
+            id: id,
+            chainId: chainId,
+            precision: UInt16(entity.precision),
+            icon: entity.icon,
+            priceId: entity.priceId,
+            price: entity.price as Decimal?
+        )
+    }
+
+    func populate(
+        entity: CDAsset,
+        from model: AssetModel,
+        using _: NSManagedObjectContext
```

Comment:
unused parametr

Reply:
this is protocol method implementation, sorry